modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/prepare/IgniteSqlToRelConvertor.java

```diff
@@ -322,4 +324,44 @@ private RelNode convertMerge(SqlMerge call) {
                 relBuilder.build(), LogicalTableModify.Operation.MERGE,
                 targetColumnNameList, null, false);
     }
+
+    // =========================================================
```

Comment:
I suggest create a ticket with a patch for Calcite and add mention the ticket here

Reply:
done

https://issues.apache.org/jira/browse/IGNITE-22755